pkg/webhook/webhook.go

```diff
@@ -0,0 +1,103 @@
+package webhook
+
+import (
+	"fmt"
+	"net/http"
+
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	configv1 "github.com/openshift/api/config/v1"
+	mcfginformersv1alpha1 "github.com/openshift/client-go/machineconfiguration/informers/externalversions/machineconfiguration/v1alpha1"
+	mcfglistersv1alpha1 "github.com/openshift/client-go/machineconfiguration/listers/machineconfiguration/v1alpha1"
+	"github.com/openshift/library-go/pkg/operator/configobserver/featuregates"
+)
+
+// NewConfig returns a new ServerConfig.
+func NewConfig(port int, certDir string) *Config {
+	return &Config{
+		Port:    fmt.Sprintf(":%d", port),
+		CertDir: certDir,
+	}
+}
+
+// Config is the configuration for the webhook server.
+type Config struct {
+	Port    string
+	CertDir string
+}
+
+// Server is a webhook server.
+type Server struct {
+	mux                  *http.ServeMux
+	cfg                  *Config
+	imageSetLister       mcfglistersv1alpha1.PinnedImageSetLister
+	imageSetListerSynced cache.InformerSynced
+	fgAccessor           featuregates.FeatureGateAccess
+}
+
+// NewServer returns a new webhook server instance which uses the controller
+// runtime admission package in standalone mode.
+func NewServer(
+	cfg *Config,
+	imageSetInformer mcfginformersv1alpha1.PinnedImageSetInformer,
+	fgAccessor featuregates.FeatureGateAccess,
+) *Server {
+	w := &Server{
+		mux:        http.NewServeMux(),
+		fgAccessor: fgAccessor,
+	}
+	w.imageSetLister = imageSetInformer.Lister()
+	w.imageSetListerSynced = imageSetInformer.Informer().HasSynced
+	return w
+}
+
+// Run starts the webhook server.
+func (s *Server) Run(stopCh <-chan struct{}) {
+	fg, err := s.fgAccessor.CurrentFeatureGates()
+	if err != nil {
+		klog.Errorf("Could not get feature gate: %v", err)
+		return
+	}
+	if !fg.Enabled(configv1.FeatureGatePinnedImages) {
```

Comment:
here is the fg. Still, shouldn't we move this check so we never access the API group?